tests/unittests/MLTest.cpp

```diff
@@ -205,6 +207,79 @@ TEST_P(MLTest, learnXor) {
   }
 }
 
+/// Learn the logarithmic function
+TEST_P(MLTest, learnLog) {
+  unsigned numInputs = 500;
+  unsigned batchSize = 10;
+  EE_.getConfig().learningRate = 0.05;
+  EE_.getConfig().batchSize = batchSize;
+
+  auto &mod = EE_.getModule();
+  Function *F = mod.createFunction("main");
+  F->setName("learnLog");
+
+  auto *A =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "A", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+  auto *Ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "Ex", VisibilityKind::Public,
+                         Variable::TrainKind::None);
+
+  Node *O = F->createFullyConnected("fc1", A, 5);
+  O = F->createTanh("tanh1", O);
+  O = F->createFullyConnected("fc2", O, 6);
+  O = F->createTanh("tanh2", O);
+  O = F->createFullyConnected("fc3", O, 1);
+  O = F->createTanh("tanh3", O);
+  O = F->createRegression("reg", O, Ex);
+  auto *result = F->createSave("ret", O);
+
+  //Set the training set 
+  Tensor trainingSet(ElemKind::FloatTy, {numInputs, 1});
+  Tensor trainingLabels(ElemKind::FloatTy, {numInputs, 1});
+
+  auto TS = trainingSet.getHandle<>();
+  auto TL = trainingLabels.getHandle<>();
+
+  float LO = 0.75;
+  float HI = 1.5;
+  srand (static_cast <unsigned> (time(0)));
```

Comment:
Why are you using rand and srand instead of the Glow RNG?

Reply:
Changed to use Glow RNG